src/ast/passes/clang_build.cpp

```diff
@@ -217,22 +220,36 @@ static Result<> build(CompileContext &ctx,
 
 ast::Pass CreateClangBuildPass()
 {
-  return ast::Pass::create("ClangBuilder",
-                           [](CompileContext &ctx,
-                              ast::Imports &imports) -> Result<BitcodeModules> {
-                             BitcodeModules result;
-
-                             // For each of the source files in the imports, we
-                             // build it and turn it into a bitcode file.
-                             for (auto &[name, obj] : imports.c_sources) {
-                               auto ok = build(ctx, name, obj, imports, result);
-                               if (!ok) {
-                                 return ok.takeError();
-                               }
-                             }
-
-                             return result;
-                           });
+  return ast::Pass::create(
+      "ClangBuilder",
+      [](BPFtrace &bpftrace,
+         CompileContext &ctx,
+         ast::Imports &imports) -> Result<BitcodeModules> {
+        BitcodeModules result;
+
+        // Construct our kernel headers.
+        std::string vmlinux_h;
+        if (!imports.c_sources.empty()) {
+          vmlinux_h = bpftrace.btf_->c_def();
```

Comment:
Looking at this confused me and I think it has to do with the `c_def` function which returns the vmlinux btf if no set is provided. How about changing that a bit where we expose a method e.g. `vmlinux_defs()` that you can call here and that `c_def` can call as well?

Reply:
> Everything is returning the vmlinux defs

It seems like the current one will only return the defs in `btf_set_` if they exist, which could be just module BTF right instead of the whole vmlinux BTF - or am I misunderstanding?

Reply:
Candidly, the`BTF` object is a bit of a mess and the way the source there is selected makes no sense to me. I believe that the module is likely a rebased on top of the full vmlinux, I'm not actually sure what will get dumped.

I'm trying to avoid going in and fixing too much there because it will quickly become yak shaving. My attempt here is to do the bare minimum to make it work for now, but I will test some cases with modules and see if it falls over.

Reply:
No worries, this conversation is no-blocking :)

Reply:
There's going to be a lot of churn in the types, so I'm just going to submit this one to unblock the USDT work at least